spec/unit/hanami/middleware/node_spec.rb

```diff
@@ -27,11 +27,23 @@
 
   describe "#get" do
     context "when segment is found" do
-      it "returns the node" do
-        segment = "foo"
-        subject.put(segment)
+      context "and segment is defined as symbol" do
+        it "returns the node" do
+          segment = "foo"
+          dynamic_segment = ":bar"
+          subject.put(dynamic_segment)
+
+          expect(subject.get(segment)).to be_kind_of(described_class)
+        end
```

Comment:
I am not sure about this test, and test from line 41 (old line 30). I am not sure, because when segment doesn't exists It returns also object of Node class (self)